src/tools/tests/storage-key-recipe-resolver-test.ts

```diff
@@ -312,6 +312,27 @@ describe('recipe2plan', () => {
         assert.isTrue(it.isResolved());
       }
     });
+    it('forces resolution of type variables', async () => {
+      const manifest = await Manifest.parse(`\
+    schema Thing 
+      name: Text
+     
+    particle Writer
+       data: writes [Thing {name}] // Could also be a singleton
```

Comment:
you can inline the schema and remove the explicit schema declaration above:
`data: writes [Thing {name: Text}]`